modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala

```diff
@@ -205,11 +253,17 @@ object ElasticRequest {
     routing: Option[Routing]
   ) extends ElasticRequest[Option[Document], GetById]
 
-  private[elasticsearch] final case class GetByQueryRequest(
+  private[elasticsearch] final case class SearchRequest(
     index: IndexName,
-    query: ElasticQuery[_],
+    query: Option[ElasticQuery[_]],
```

Comment:
We might want to think about separating into 3 case classes to represent only valid combinations, something like:
- SearchRequest
- AggregationRequest
- SearchWithAggregationRequest

Reply:
Possibly having then the API:
- search(query)
- aggregate(aggregation)
- search(query).aggregate(aggregation)